.drone.star

```diff
@@ -2234,12 +2224,19 @@ def genericCachePurge(name, cache_key):
     }
 
 def genericBuildArtifactCache(ctx, name, action, path):
-    name = "%s_build_artifact_cache" % (name)
-    cache_key = "%s/%s/%s" % (ctx.repo.slug, ctx.build.commit + "-${DRONE_BUILD_NUMBER}", name)
+    cache_path = "%s/%s/%s" % ("cache", ctx.repo.slug, ctx.build.commit + "-${DRONE_BUILD_NUMBER}")
+
     if action == "rebuild" or action == "restore":
-        return genericCache(name, action, [path], cache_key)
+        name = "%s_build_artifact_cache" % (name)
+        return genericCache(name, action, [path], cache_path)
+
     if action == "purge":
-        return genericCachePurge(name, cache_key)
+        # cached items which are not purged due to build cancel are purged on nightly
+        if ctx.build.event == "cron":
+            flush_path = "%s/%s" % ("cache", ctx.repo.slug)
```

Comment:
I there is an ordinary build that happens to be running when the nightly cron build runs, then what is going to happen? The cache of that ordinary PR build might unexpectedly disappear.

We could set the `flush_age` (line 2209) to `1` day - "random rubbish" will sit for an extra day before being cleaned up, but we avoid accidentally breaking a running PR  build.